lib/macho/macho_file.rb

```diff
@@ -172,10 +170,14 @@ def command(name)
 
     # Add a new load command to the Mach-O.
     # @param lc [MachO::LoadCommand] the load command being added
+    # @param repopulate [Boolean] whether or not to repopulate the instance fields
     # @return [void]
     # @raise [MachO::HeaderPadError] if the new command exceeds the header pad buffer
     # @note This is public, but methods like {#add_rpath} should be preferred.
-    def add_command(lc)
+    #  Setting `repopulate` to false **will leave the instance in an
+    #  inconsistent state** unless {#populate_fields} is called **immediately**
+    #  afterwards.
+    def add_command(lc, repopulate = true)
```

Comment:
> Making repopulate a simple boolean argument is convenient, but can be difficult to understand on the call site

I was thinking the same thing. Having it as an option is certainly more obvious, and gives us future flexibility :+1: